### PR TITLE
feature: extent and test pixel-meter conversion

### PR DESCRIPTION
**Pull request by Dummyrunner**

### CODE REVIEW

File: src/general_physics.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+from dataclasses import dataclass
+from src.common_constants import CommonConstants
+from src.vec2d import Vec2d
+
+
+@dataclass(frozen=True)
```
Comment (beneblubb) [inline src/general_physics.py:6]: Why as dataclass?

File: src/general_physics.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+class GeneralPhysics:
```
Comment (beneblubb) [inline src/general_physics.py:7]: Is calculating pixels a physics problem? Why not call it "size convertion" or similar?

File: src/general_physics.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    pixel_to_meter_ratio: float = CommonConstants.PIXEL_TO_METER
+    meter_to_pixel_ratio: float = CommonConstants.METER_TO_PIXEL
+    timestep_size: float = 1 / 60
+
+    def pixel_to_meter(*args) -> float:
+        if type(args[0]) == float and len(args) == 1:
+            res = args[0] * GeneralPhysics.pixel_to_meter_ratio
+        elif type(*args) == Vec2d:
+            res = GeneralPhysics.pixel_to_meter_ratio * args[0]
```
Comment (beneblubb) [inline src/general_physics.py:16]: Why changing the order of the factors? Makes it more difficult to read. Or has this a reason I don't get?

**Review by beneblubb — APPROVED** [review]
Great!
No functional, only cosmetic questions.

File: src/general_physics.py (hunk 2 above)
Comment (Dummyrunner) [reply src/general_physics.py]: It Comes into Play when doing physical calculations, to make these better to grasp intuitively. Apart from that, we could also set everything in pixels

**Review by beneblubb — APPROVED** [review]
Looks good, only some little lines a noob like me doesn't quite understand

File: src/general_physics.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+from src.common_constants import CommonConstants
+from src.vec2d import Vec2d
+
+
+pixel_to_meter_ratio: float = CommonConstants.PIXEL_TO_METER
+meter_to_pixel_ratio: float = CommonConstants.METER_TO_PIXEL
+timestep_size: float = 1 / 60
```
[the remark this answers is not in the document]
Comment (Dummyrunner) [reply src/general_physics.py:7]: this is just an alias for improving readability

File: src/landing_game_object.py [hunk 5]
```diff
@@ -10,17 +10,33 @@ class LandingGameObject(pygame.sprite.Sprite):
...
         self,
         image: pygame.surface,
-        pos: Vec2d = Vec2d(),
+        pos_pixel: Vec2d = Vec2d(),
     ) -> None:
         super().__init__()
         self.rect: pygame.Rect = pygame.Surface.get_rect(image)
-        self.pos = pos
-        self.rect.center = self.pos
+        self.rect.center = Vec2d(pos_pixel)
```
[the remark this answers is not in the document]
Comment (Dummyrunner) [reply src/landing_game_object.py:17]: The change in overlay from using `__dict__`  to using `dir()` and `getattr(obj, attr_name) ` as a subject could be integrated in PR #21 .
Maybe we also could think about giving a warning and some fallback value if a requested overlay attribute is not avaliable instead of letting the program crash.

File: src/landing_game_object.py [hunk 6]
```diff
@@ -10,17 +10,33 @@ class LandingGameObject(pygame.sprite.Sprite):
...
+    def set_pos(self, pos_pixel: Vec2d) -> None:
+        """set position in pixel standard coordinate system
+
+        Args:
+            pos_pixel (Vec2d): position in pixels
+        """
+        self.rect.center = Vec2d(pos_pixel)
+        self.add_pos_to_dict()
+
+    def add_pos_to_dict(self):
```
[the remark this answers is not in the document]
Comment (Dummyrunner) [reply src/landing_game_object.py:41]: The change in overlay from using `__dict__`  to using `dir()` and `getattr(obj, attr_name) ` as a subject could be integrated in PR #21 .
Maybe we also could think about giving a warning and some fallback value if a requested overlay attribute is not avaliable instead of letting the program crash.

File: src/landing_game_object.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (beneblubb) [reply src/landing_game_object.py:17]: Arr!

File: src/general_physics.py (hunk 3 above)
Comment (Dummyrunner) [reply src/general_physics.py]: Aligned it.

File: src/general_physics.py (hunk 2 above)
Comment (beneblubb) [reply src/general_physics.py]: Im still curious how translating between real and digital world will work, but i'm sure you have a good solution in your mind.

File: src/landing_game_object.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (beneblubb) [reply src/landing_game_object.py:41]: Still don't understand your answer - why "The change in overlay..."? My Question is about adding a .pos instead of just using .rect.center. Same functionality, just a few more characters. 
Is there a reason I still don't get?
